yarl/_url.py

```diff
@@ -435,6 +450,21 @@ def raw_authority(self):
         """
         return self._val.netloc
 
+    def _get_default_port(self):
+        port = None
+        if self.scheme:
+            port = DEFAULT_PORTS.get(self.scheme)
+            if port is None:
+                with suppress(OSError):
+                    port = socket.getservbyname(self.scheme)
+        return port
```

Comment:
Let's reduce a few levels of nesting:
```suggestion
        if not self.scheme:
            return None

        with suppress(KeyError):
            return DEFAULT_PORTS[self.scheme]

        with suppress(OSError):
            return socket.getservbyname(self.scheme)

        return None
```